src/Compilers/CSharp/Portable/Lowering/Instrumentation/DebugInfoInjector.cs

```diff
@@ -431,5 +431,11 @@ public override BoundExpression InstrumentSwitchStatementExpression(BoundStateme
             return AddConditionSequencePoint(base.InstrumentSwitchStatementExpression(original, rewrittenExpression, factory), original.Syntax, factory);
         }
 
+        public override BoundStatement InstrumentPatternSwitchBindCasePatternVariables(BoundStatement bindings)
+        {
+            // Mark the code that binds pattern variables to their values as hidden.
+            // We do it to tell that this is not a part of previous statement.
+            return new BoundSequencePoint(null, base.InstrumentPatternSwitchBindCasePatternVariables(bindings));
```

Comment:
>return new BoundSequencePoint(null, base.InstrumentPatternSwitchBindCasePatternVariables(bindings)); [](start = 12, length = 100)

If we don't plan to instrument this place, it would be fine to inject hidden sequence points in the local rewriter and keep the instrumentation API smaller. We are doing this in several places.